docs/fastdds/library_overview/includes/architecture.rst

```diff
@@ -0,0 +1,107 @@
+  distributed systems.
+* **Fast DDS layer**.
+  Robust implementation of the DDS communications middleware.
+  It allows the development of one or more DDS domains in which DomainParticipants within the same domain
+  exchange messages by publishing under a domain topic and subscribing to a it.
+* **RTPS layer**.
+  Implementation of the Real-Time Publish-Subscribe (RTPS) protocol for interoperability with DDS applications.
+  This layer acts as the DDS abstraction layer of the transport layer.
+* **Transport Layer**.
+  Fast DDS can be implemented over various transport protocols such as unreliable transport protocols (UDP), reliable
+  transport protocols (TCP), and shared memory transport protocols (SHM).
+
+.. figure:: /01-figures/fast_dds/library_overview/library_overview.svg
+  :align: center
+
+  Fast DDS layer model architecture
+
+The elements that comprise each of the layers shown in the figure above are explained below.
+
+DDS Layer
+^^^^^^^^^
+
+Several key elements for communication are defined in the DDS layer of Fast DDS.
+The user will create these elements in his/her application, thus incorporating DDS application elements and creating a
+data-centric communication system.
+Fast DDS, following the DDS specification, defines these elements involved in communication as **Entities**.
+
+* **DomainId_t**.
+  uint32_t data type which identifies the DDS domain.
+  Each DomainParticipant will have an assigned DDS domain, so that DomainParticipants in the same domain can
+  communicate, as well as isolate communications between DDS domains.
+  This value must be given by the application developer when creating the DomainParticipants.
+* **DomainParticipants**.
+  Object containing other DDS entities such as publishers, subscribers, topics and multitopics.
+  It is the entity that allows the creation of the previous entities it contains, as well as the configuration of their
+  behavior.
+* **Publisher**.
+  It is the entity that creates and configures the DataWriter entities it contains.
+  It may contain one or more DataWriter entities.
+* **DataWriter**.
+  It is the entity in charge of publishing messages.
+  The user must provide a Topic when creating this entity which will be the Topic under which the data will be
+  published.
+  Publication is done by writing the data-objects as a change in the DataWriterHistory.
+* **DataWriterHistory**.
+  It contains the data objects that are changed by the DataWriter and that are going to be served to the DataReader
+  that subscribes to the topic under which the DataWriter publishes.
+* **Subscriber**.
+  It is the entity that creates and configures the DataReader entities it contains.
+  It may contain one or more DataReader entities.
+* **DataReader**.
+  It is the entity that subscribes to the topics for the reception of publications.
+  The user must provide a subscription Topic when creating this entity.
+  A DataReader receives the messages as changes in its HistoryDataReader.
```

Comment:
Similar to publisher. The subscriber subscribes to a topic using a DataReader, which read the data from the transport